crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -0,0 +1,803 @@
+use tedge_actors::SimpleMessageBox;
+use tedge_api::OperationStatus;
+use tedge_file_system_ext::FsWatchEvent;
+use tedge_http_ext::HttpRequest;
+use tedge_http_ext::HttpRequestBuilder;
+use tedge_http_ext::HttpResponseExt;
+use tedge_http_ext::HttpResult;
+use tedge_mqtt_ext::MqttMessage;
+use tedge_mqtt_ext::Topic;
+use tedge_utils::paths::PathsError;
+use time::OffsetDateTime;
+
+use super::error::LogRetrievalError;
+use super::json::LogRequestPayload;
+use super::json::LogResponsePayload;
+use super::LogManagerConfig;
+use super::LogPluginConfig;
+
+fan_in_message_type!(LogInput[MqttMessage, FsWatchEvent] : Debug);
+fan_in_message_type!(LogOutput[MqttMessage]: Debug);
+
+pub struct LogManagerActor {
+    config: LogManagerConfig,
+    plugin_config: LogPluginConfig,
+    mqtt_publisher: LoggingSender<MqttMessage>,
+    messages: SimpleMessageBox<LogInput, NoMessage>,
+    http_proxy: ClientMessageBox<HttpRequest, HttpResult>,
+}
+
+impl LogManagerActor {
+    pub fn new(
+        config: LogManagerConfig,
+        plugin_config: LogPluginConfig,
+        mqtt_publisher: LoggingSender<MqttMessage>,
+        messages: SimpleMessageBox<LogInput, NoMessage>,
+        http_proxy: ClientMessageBox<HttpRequest, HttpResult>,
+    ) -> Self {
+        Self {
+            config,
+            plugin_config,
+            mqtt_publisher,
+            messages,
+            http_proxy,
+        }
+    }
+
+    pub async fn process_mqtt_message(
+        &mut self,
+        message: MqttMessage,
+    ) -> Result<(), anyhow::Error> {
+        if self.config.logtype_reload_topic.accept(&message) {
+            self.reload_supported_log_types().await?;
+        } else if self.config.logfile_request_topic.accept(&message) {
+            match LogRequestPayload::try_from(message.clone()) {
+                Ok(request) => {
+                    info!("Log request received: {request:?}");
+                    self.handle_logfile_request_operation(&message.topic, &request)
+                        .await?;
+                }
+                Err(_) => {
+                    error!("Incorrect log request payload: {message:?}");
+                }
+            }
+        } else {
+            error!(
+                "Received unexpected message on topic: {}",
+                message.topic.name
+            );
+        }
+
+        Ok(())
+    }
+
+    fn read_log_content(
+        logfile: &Path,
+        mut line_counter: usize,
+        max_lines: usize,
+        filter_text: &Option<String>,
+    ) -> Result<(usize, String), LogRetrievalError> {
+        if line_counter >= max_lines {
+            Err(LogRetrievalError::MaxLines)
+        } else {
+            let mut file_content_as_vec = VecDeque::new();
+            let file = std::fs::File::open(logfile)?;
+            let file_name = format!(
+                "filename: {}\n",
+                logfile.file_name().unwrap().to_str().unwrap() // never fails because we check file exists
```

Comment:
This is not true as the `OsStr` filename might not be valid UTF-8.

It might be simpler to use [`camino::Utf8Path`](https://docs.rs/camino/latest/camino/struct.Utf8Path.html)